tools/pymanifold.cpp

```diff
@@ -0,0 +1,169 @@
+             return self.Rotate(xDegrees, yDegrees, zDegrees);
+           }, py::arg("xDegrees"), py::arg("yDegrees"), py::arg("zDegrees"))
+      .def("export",
+           [](Manifold &self, std::string name) { exportManifold(self, name); },
+        py::arg("filename"),
+           "Export the manifold object to file, where the file type is determined from file extension.")
+      .def("warp",
+           [](Manifold self, const std::function<Float3(Float3)> &f) {
+             return self.Warp([&f](glm::vec3 &v) {
+               Float3 fv = f(std::make_tuple(v.x, v.y, v.z));
+               v.x = std::get<0>(fv);
+               v.y = std::get<1>(fv);
+               v.z = std::get<2>(fv);
+             });
+           }, py::arg("f"))
+      .def("refine", [](Manifold self, int n) { return self.Refine(n); }, py::arg("n"))
+      .def_static(
+          "FromMesh",
+          [](py::array_t<float> &vertPos, py::array_t<int> &triVerts) {
+            auto vertPos_view = vertPos.unchecked<2>();
+            auto triVerts_view = triVerts.unchecked<2>();
+            if (vertPos_view.shape(1) != 3)
+              throw std::runtime_error("Invalid vertex position shape");
+            if (triVerts_view.shape(1) != 3)
+              throw std::runtime_error("Invalid triangle vertices shape");
+            std::vector<glm::vec3> vertPos_vec(vertPos_view.shape(0));
+            std::vector<glm::ivec3> triVerts_vec(triVerts_view.shape(0));
+            for (int i = 0; i < vertPos_view.shape(0); i++)
+              vertPos_vec[i] = {vertPos_view(i, 0), vertPos_view(i, 1),
+                                vertPos_view(i, 2)};
+            for (int i = 0; i < triVerts_view.shape(0); i++)
+              triVerts_vec[i] = {triVerts_view(i, 0), triVerts_view(i, 1),
+                                 triVerts_view(i, 2)};
+            return Manifold({vertPos_vec, triVerts_vec});
+          }, py::arg("vertPos"), py::arg("triVerts"))
+      .def_static("Tetrahedron", []() { return Manifold::Tetrahedron(); })
+      .def_static(
+          "Cube",
+          [](float x, float y, float z,
+             bool center =
+                 false) { return Manifold::Cube(glm::vec3(x, y, z), center); },
+          py::arg("x"), py::arg("y"), py::arg("z"), py::arg("center") = false)
+      .def_static(
+          "Cylinder",
+          [](float height, float radiusLow, float radiusHigh = -1.0f,
+             int circularSegments = 0) {
+            return Manifold::Cylinder(height, radiusLow, radiusHigh,
+                                      circularSegments);
+          },
+          py::arg("height"), py::arg("radiusLow"),
+          py::arg("radiusHigh") = -1.0f, py::arg("circularSegments") = 0)
+      .def_static(
+          "Sphere",
+          [](float radius,
+             int circularSegments =
+                 0) { return Manifold::Sphere(radius, circularSegments); },
+          py::arg("radius"), py::arg("circularSegments") = 0);
+  ;
```

Comment:
intentional?

Reply:
Sorry, I walk talking about the one-line `;` on 134.

Reply:
oh, that was not intentional. deleted